lib/entry.js

```diff
@@ -3,14 +3,18 @@
 var DIRECTORY_MODE = 16877;
 
 module.exports = Entry;
-function Entry(relativePath, size, mtime) {
-  var isDirectory = relativePath.charAt(relativePath.length - 1) === '/';
+function Entry(relativePath, size, mtime, mode) {
+  if (mode === undefined) {
```

Comment:
Sorry. I don't mean to be annoying, but I have debugged too many bugs that resulted from ^^.